scipy/sparse/base.py

```diff
@@ -1019,11 +1019,10 @@ def _is_integral(dtype):
             return (inter_self * (1.0 / self.shape[1])).sum(
                 axis=1, dtype=res_dtype, out=out)
 
-    def diagonal(self):
+    def diagonal(self, k=0):
         """Returns the main diagonal of the matrix
         """
```

Comment:
We're no longer returning just the main diagonal. This docstring should also define k, and maybe give a usage example? A see-also reference to numpy's diagonal would be good, too.